packages/mui-material/src/Modal/Modal.js

```diff
@@ -99,58 +110,110 @@ const Modal = React.forwardRef(function Modal(inProps, ref) {
     disableScrollLock,
     hideBackdrop,
     keepMounted,
-    onBackdropClick,
-    onClose,
-    open,
   };
 
+  const {
+    getRootProps,
+    getBackdropProps,
+    getTransitionProps,
+    portalRef,
+    isTopModal,
+    exited,
+    hasTransition,
+  } = useModal({
+    ...propsWithDefaults,
+    rootRef: ref,
+  });
+
   const ownerState = {
-    ...props,
-    ...commonProps,
+    ...propsWithDefaults,
     exited,
   };
 
+  const classes = useUtilityClasses(ownerState);
+
+  const childProps = {};
+  if (children.props.tabIndex === undefined) {
+    childProps.tabIndex = '-1';
+  }
+
+  // It's a Transition like component
+  if (hasTransition) {
+    const { onEnter, onExited } = getTransitionProps();
+    childProps.onEnter = onEnter;
+    childProps.onExited = onExited;
+  }
+
   const RootSlot = slots?.root ?? components.Root ?? ModalRoot;
   const BackdropSlot = slots?.backdrop ?? components.Backdrop ?? BackdropComponent;
 
   const rootSlotProps = slotProps?.root ?? componentsProps.root;
   const backdropSlotProps = slotProps?.backdrop ?? componentsProps.backdrop;
 
+  const rootProps = useSlotProps({
+    elementType: RootSlot,
+    externalSlotProps: rootSlotProps,
+    externalForwardedProps: other,
+    getSlotProps: getRootProps,
+    additionalProps: {
+      ref,
+      as: component,
+    },
+    ownerState,
+    className: clsx(
+      className,
+      rootSlotProps?.className,
+      classes?.root,
+      !ownerState.open && ownerState.exited && classes?.hidden,
+    ),
+  });
+
+  const backdropProps = useSlotProps({
+    elementType: BackdropSlot,
+    externalSlotProps: backdropSlotProps,
+    additionalProps: BackdropProps,
+    getSlotProps: (otherHandlers) => {
+      return getBackdropProps({
+        ...otherHandlers,
+        onClick: (e) => {
```

Comment:
We don't allow variable shorthands:

```suggestion
        onClick: (event) => {
```